engine/src/main/java/org/hibernate/search/query/dsl/sort/SortAdditionalSortFieldContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+import org.apache.lucene.search.SortField;
```

Comment:
I didn't follow the entire discussion, so it may have been discussed before: is it right that we don't fully abstract from Lucene here? I'd have kind of hoped we'd find a generic approach working for both Lucene, ES (and Solr in the future). I.e. should we continue to expose Lucene specifics or rather stop doing it in new APIs? I'd strongly lean towards the latter.